pkg/await/apps_deployment.go

```diff
@@ -528,29 +546,65 @@ func (dia *deploymentInitAwaiter) checkReplicaSetStatus() {
 	var rawReadyReplicas interface{}
 	var readyReplicas int64
 	var readyReplicasExists bool
+	var unavailableReplicas int64
+	var expectedNumberOfUpdatedReplicas bool
 	// extensions/v1beta1/ReplicaSet does not include the "readyReplicas" status for rollouts,
 	// so use the Deployment "readyReplicas" status instead.
 	// Note: We must use the input apiVersion rather than the Deployment watch Event we're processing here, because
 	// the Progressing status field will not be present if the Deployment was created with the `extensions/v1beta1` API,
 	// regardless of what the Event apiVersion says.
-	statusUnavailable := dia.config.createAwaitConfig.currentInputs.GetAPIVersion() == "extensions/v1beta1"
-	if statusUnavailable {
+	extensionsv1beta1API := dia.config.createAwaitConfig.currentInputs.GetAPIVersion() == "extensions/v1beta1"
+	if extensionsv1beta1API {
 		rawReadyReplicas, readyReplicasExists = openapi.Pluck(dia.deployment.Object, "status", "readyReplicas")
 		readyReplicas, _ = rawReadyReplicas.(int64)
 	} else {
 		rawReadyReplicas, readyReplicasExists = openapi.Pluck(rs.Object, "status", "readyReplicas")
 		readyReplicas, _ = rawReadyReplicas.(int64)
 	}
 
+	if rawUpdatedReplicas, ok := openapi.Pluck(dia.deployment.Object, "status", "updatedReplicas"); ok {
+		updatedReplicas, _ := rawUpdatedReplicas.(int64)
+		expectedNumberOfUpdatedReplicas = updatedReplicas == specReplicas
+	}
+
+	// Check replicas status, which is present on all apiVersions of the Deployment resource.
+	// Note that this status field does not appear immediately on update, so it's not sufficient to
+	// determine readiness by itself.
+	rawReplicas, replicasExists := openapi.Pluck(dia.deployment.Object, "status", "replicas")
+	replicas, _ := rawReplicas.(int64)
+	tooManyReplicas := replicasExists && replicas > specReplicas
+
+	// Check unavailableReplicas status, which is present on all apiVersions of the Deployment resource.
+	// Note that this status field does not appear immediately on update, so it's not sufficient to
+	// determine readiness by itself.
+	unavailableReplicasPresent := false
+	if rawUnavailableReplicas, ok := openapi.Pluck(
+		dia.deployment.Object, "status", "unavailableReplicas"); ok {
+		unavailableReplicas, _ = rawUnavailableReplicas.(int64)
+
+		unavailableReplicasPresent = unavailableReplicas != 0
+	}
+
```

Comment:
I'm not sure why this chunk of added code should not be behind `if extensionsv1beta1API {` also?

Reply:
Will do